structured_external_storage_all/models/ir_attachment.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# © 2018 Sunflower IT (http://sunflowerweb.nl)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import models, fields, api, _
+
+
+class Attachment(models.Model):
+    _inherit = 'ir.attachment'
+
+    def create(self, vals):
+        res = super(Attachment, self).create(vals)
+        res.create_metadata()
+        return res
+
+    @api.multi
+    def create_metadata(self):
+        """Use sequence to obtain the rule with high priority"""
+        sync_rule_obj = self.env['ir.attachment.sync.rule']
+        for attachment in self:
+            # check if res_model is in the define rules.
+            all_matching_rules = sync_rule_obj.search([
+                ('source_model.object', '=', attachment.res_model)
+            ], order='sequence asc')
+            selected_matching_rules = []
+            for rule in all_matching_rules:
+                previous_rules_with_same_loc_and_lower_seq = [
+                    previous_rule for previous_rule in selected_matching_rules
+                    if previous_rule.location_id == rule.location_id
```

Comment:
I propose changing this to `sync_type`, I thought a location was Dropbox but we will have locations for each Dropbox subfolder